### PR TITLE
Add missing 'not in' predicate support to `read_parquet`

### PR DESCRIPTION
**Pull request by rjzamora**
`dd.read_parquet` does not currently support filters containing `"not in"` predicates. In fact, if the user passes in something like `filters=[("B", "not in", (1, 2))]`, Dask will return an empty `dd.DataFrame`!

This PR adds basic `"not in"` support to address this bug/oversight.

### CODE REVIEW

File: dask/dataframe/io/parquet/core.py [hunk 1]
```diff
@@ -1394,6 +1396,8 @@ def apply_conjunction(parts, statistics, conjunction):
                         and max >= value
                         or operator == "in"
                         and any(min <= item <= max for item in value)
+                        or operator == "not in"
+                        and not any(min == max == item for item in value)
```
Comment (hendrikmakait) [inline dask/dataframe/io/parquet/core.py:1400]: Should we raise on unhandled operators instead of silently ignoring them?

**Review by hendrikmakait — APPROVED** [review]
Thanks for fixing this, @rjzamora! I have one question for clarification/food for thought directed at avoiding similar issues.

File: dask/dataframe/io/parquet/core.py (hunk 1 above)
Comment (rjzamora) [reply dask/dataframe/io/parquet/core.py:1400]: Ah, yes. I agree that the problem is that we were silently filtering for an "unsupported" operator.  I'll try to make the list of supported operators more explicit.

Comment (hendrikmakait) [reply dask/dataframe/io/parquet/core.py:1420]: :tada:

**Review by jrbourbeau — APPROVED** [review]
Thanks @rjzamora

**Pull request merged**